### PR TITLE
Generate Organizers' logged in Token

### PR DESCRIPTION
**Pull request by miajhaochen**
**Dependency #1** 

### DONE
1. update `config.js` to make it more general
2. install `jsonwebtoken ` to generate jwt token as a proof that the user has logged in
3. in api `POST /api/login`, generate jwt token after the organizer has given the correct pair of email and password 
4. write a function `checkAuth` used to check whether the request can provide correct jwt token to execute the api
> a simple usage is in api `POST /api/event`

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -14,7 +14,13 @@ var api = require('./routes/api');
 var app = express();
 var port = process.env.PORT || 3000;
 
+app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
+app.use(function(req, res, next) {
+  res.header("Access-Control-Allow-Origin", "http://localhost:3000");
```
Comment (miajhaochen) [inline server.js]: should be fixed in #5

File: routes/api.js [hunk 2]
```diff
@@ -93,10 +100,14 @@ router.post('/login', [
...
   // check password
-  const hashedPwd = ret.data[0].password;
+  const user = ret.data[0];
+  const hashedPwd = user.password;
   if (!bcrypt.compareSync(password, hashedPwd))
     return res.json({success: false, message: "The password is incorrect."});
-  return res.json(ret);
+  // issue token
+  const payload = {usersId: user.usersId, email: user.email};
+  const token = jwt.sign(payload, jwtSecret, {expiresIn: '2h'});
```
Comment (bwan2) [inline routes/api.js]: Old tokens can be used if an organizer logs in multiple times. Should there only be a 'lastest' token that an organizer can use?

File: utils/auth.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken');
+const config = require('../config.js');
+const env = process.env.NODE_ENV || 'local';
+const jwtSecret = config[env].jwtSecret;
+
+module.exports.checkAuth = (req, res, next) => {
+  const token = req.body.token;
+  if (!token) {
+    res.status(401).send('Unauthorized: No token provided');
```
Comment (bwan2) [inline utils/auth.js]: should this be JSON to be consistent?

File: utils/auth.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  } else {
+    jwt.verify(token, jwtSecret, function(err, payload) {
+      if (err) {
+        res.status(401).send('Unauthorized: Invalid token');
```
Comment (bwan2) [inline utils/auth.js]: should this be JSON to be consistent?

**Review by bwan2 — CHANGES_REQUESTED** [review]
1. Old tokens can be used by organizers if they logged in multiple times. Should we limit only the latest token?
2. Sending JSON back for error messages to be consistent.

File: routes/api.js (hunk 2 above)
Comment (miajhaochen) [reply routes/api.js]: fix this after the database is chosen loooooool

**Pull request merged**